src/Metadata/PropertyTypeArray.php

```diff
@@ -30,6 +36,9 @@ public function __toString(): string
             return 'array';
         }
         $array = $this->isHashmap() ? '[string]' : '[]';
+        if ($this->isCollection) {
+            $array .= '|Collection';
```

Comment:
i think that when we don't know the type, we should still return `array|Collection` a few lines above